supervision/detection/line_zone.py

```diff
@@ -233,21 +241,65 @@ def trigger(self, detections: Detections) -> Tuple[np.ndarray, np.ndarray]:
                 continue
 
             tracker_state = has_any_left_trigger[i]
-            if tracker_id not in self.tracker_state:
-                self.tracker_state[tracker_id] = tracker_state
+            if tracker_id not in self.crossing_state:
+                self.crossing_state[tracker_id] = (
+                    class_ids,
+                    deque([tracker_state], maxlen=self.max_linger),
+                )
+                continue
+
+            crossing_state_class_ids, crossing_state = self.crossing_state[tracker_id]
+            prev_frame_tracker_state = crossing_state[-1]
+            if self.max_linger == 1 and prev_frame_tracker_state == tracker_state:
                 continue
 
-            if self.tracker_state.get(tracker_id) == tracker_state:
+            crossing_in_progress = (
+                crossing_state.count(True) != 0 and crossing_state.count(False) != 0
+            )
+            crossing_state.appendleft(tracker_state)
+            all_on_same_side = crossing_state.count(not tracker_state) == 0
+            if class_ids:
+                if len(class_ids) != len(crossing_state_class_ids) or not all(
+                    class_ids == crossing_state_class_ids
+                ):
+                    self.crossing_state[tracker_id] = (class_ids, tracker_state)
```

Comment:
Inconsistency here: second element was `dequeue`, but adding tracker_state which is `bool`.

(I'll figure it out, no need to reply :)